### PR TITLE
Set up redux infrastructure

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

Sets up the initial redux infrastructure for maintaining frontend state. We choose to use redux / react-redux / redux-toolkit libraries since it is the consistent way that core OSD + other plugins use, and is the standard across modern react applications. It also doesn't add any dependencies since we consume them from core OSD.
- sets up global redux store via `configureStore`
- sets up basic workspace reducer to process workspace-related state updates via `createSlice`
- wraps the entire app within a `Provider` so React components have access to the store
- sets up sample cmds for future reference as ways to access and set the store in the base `WorkflowBuilder` page

Tested with some basic functionalities, including adding some dummy components, and validating if there are changes. These are using the `components` and `isDirty` vars within the `workspace` state.

[screen-capture (16).webm](https://github.com/opensearch-project/opensearch-ai-flow-dashboards/assets/62119629/56c1b67d-4251-405f-9eb0-6feefee54676)

### Issues Resolved

Makes progress on #8 & #10

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/store/reducers/workspace_reducer.ts
Comment (owaiskazi19) [inline public/store/reducers/workspace_reducer.ts:13]: Silly questions: But don't we have to treat this as an actionType?

File: public/store/reducers/workspace_reducer.ts
Comment (ohltyler) [reply public/store/reducers/workspace_reducer.ts:22]: We can - prefer to leave for now and as more implementation comes we can decide on if we want a pattern like this, or what you're suggesting. It may get complex where one action may be updating several different state vars.

File: public/store/reducers/workspace_reducer.ts
Comment (ohltyler) [reply public/store/reducers/workspace_reducer.ts:13]: `createSlice` takes care of that for us, where we don't need to create action strings at all. See https://redux.js.org/tutorials/fundamentals/part-8-modern-redux#writing-slices

File: public/pages/workflow_builder/workflow_builder.tsx
Comment (ohltyler) [reply public/pages/workflow_builder/workflow_builder.tsx:21]: I don't have an explicit issue for the redux state, kind of treating it implicitly in #8 and #10. The commented out lines show how to use dispatch to call on the `setDirty`/`removeDirty`/ etc. action creators.

File: public/store/reducers/workspace_reducer.ts
Comment (owaiskazi19) [reply public/store/reducers/workspace_reducer.ts:13]: Oh! Something new for me. I'll read about it.

**Review by owaiskazi19 — APPROVED** [review]
Looks good on my end but I'll wait for @jackiehanyang review before merging it.

**Comment by jackiehanyang** [comment]
LGTM!

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**
